mobile/src/bundles/common/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,54 @@
+import CheckBox, { type CheckBoxProps } from '@react-native-community/checkbox';
+import React from 'react';
+import {
+    type Control,
+    type FieldPath,
+    type FieldValues,
+} from 'react-hook-form';
+
+import { Text, View } from '~/bundles/common/components/components';
+import { Color } from '~/bundles/common/enums/enums';
+import { useFormController } from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/styles';
+
+import { styles } from './styles';
+
+type Properties<T extends FieldValues> = CheckBoxProps & {
+    label?: string;
+    name: FieldPath<T>;
+    control: Control<T, null>;
+};
+
+const Checkbox = <T extends FieldValues>({
+    label,
+    name,
+    control,
+    ...props
+}: Properties<T>): JSX.Element => {
+    const { field } = useFormController({ name, control });
+    const { value, onChange } = field;
+
+    return (
+        <View
+            style={[
+                globalStyles.flexDirectionRow,
+                globalStyles.alignItemsFlexStart,
+                styles.container,
+            ]}
+        >
+            <CheckBox
+                value={value}
+                onValueChange={onChange}
+                tintColors={{ true: Color.PRIMARY, false: Color.INPUT }}
```

Comment:
I used your checkbox component. It works, but the "disable"  does not work. Maybe it should be written like this: const colorCheckbox = props.disabled ? Color.TEXT2 : Color.PRIMARY;

tintColors={{ true: colorCheckbox, false: Color.INPUT }}

Reply:
I've been testing "disabled."
Just when you call a component you can write 'disabled' and this property will be passed.
It is his own prop.
Like:
<Checkbox ... disabled> and it will be disabled.

Reply:
But color does not change

Reply:
![check](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/112649713/b9ef996b-eb50-4c75-8f6a-33b1be147e30)

Reply:
Well, as far as I understand.
If the checkbox is disabled, it means that it cannot be "checked". 
So it will look like this and will not respond to clicks.
<img width="164" alt="Screenshot 2023-08-27 at 20 51 33" src="https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/70234990/ccdb6434-736d-4107-86c9-a11ebaae9993">